monitoring/uss_qualifier/scenarios/astm/netrid/common/misbehavior.py

```
@@ -114,33 +160,102 @@ def _poll_unauthenticated_during_flights(
+            diagonals_m,
+            poll_func,
+        )
+
+    def _fetch_flights_from_dss(self, rect: LatLngRect):
+        # We grab all flights from the SPs (which we know how to reach by first querying the DSS).
+        # This is authenticated and is expected to succeed
+        sp_observation = rid.all_flights(
+            rect,
+            include_recent_positions=True,
+            get_details=True,
+            rid_version=self._rid_version,
+            session=self._dss.client,
+            dss_participant_id=self._dss.participant_id,
         )
 
+        mapping_by_injection_id = (
+            display_data_evaluator.map_fetched_to_injected_flights(
+                self._injected_flights, list(sp_observation.uss_flight_queries.values())
+            )
+        )
+        self.record_queries(sp_observation.queries)
+
+        return mapping_by_injection_id
+
+    def _evaluate_and_test_too_large_area_requests(
+        self,
+        rect: LatLngRect,
+    ) -> Set[str]:
+        """Queries all flights from the DSS to discover flights urls and query them using a larger area than allowed.
+
+        :returns: set of injection IDs that were encountered and tested
+        """
+
+        mapping_by_injection_id = self._fetch_flights_from_dss(rect)
+        for injection_id, mapping in mapping_by_injection_id.items():
+            self._evaluate_too_large_area(rect, injection_id, mapping)
+
+        return set(mapping_by_injection_id.keys())
+
+    def _evaluate_too_large_area(
+        self, rect: LatLngRect, injection_id: str, mapping: TelemetryMapping
+    ):
+        participant_id = mapping.injected_flight.uss_participant_id
+        flights_url = mapping.observed_flight.query.flights_url
+        session = self._dss.client
+
+        scale = LatLng(0.01, 0.001)
+        invalid_rect = rect.expanded(scale)
+        diagonal_km = geo.get_latlngrect_diagonal_km(invalid_rect)
+        if diagonal_km > self._rid_version.max_diagonal_km:
```

Why this condition? If a sanity check on the input, shouldn't it rather raise an exception? But then why expand the rect?